### PR TITLE
Spiral matrix code made generalised. Readme added

### PR DESCRIPTION
**Pull request by earora97**
**Fixes issue:**
1. Readme added with an explanation of problem and code. Added example.
2. .cpp code made generalised.
**Changes:**

### CODE REVIEW

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 1]
```diff
@@ -43,19 +43,15 @@ void spiralPrint(int a[][100],int m,int n){
...
-    for(int i=0;i<m;i++){
-        for(int j=0;j<n;j++){
-            a[i][j] = value++;
-            cout<<a[i][j]<<" ";
-        }
-        cout<<endl;
-    }
+     int m,n; ///Dimensions of 2D array
+    
+    cout << "Input the Dimensions of the 2-D array" << endl;
```
Comment (arnavb) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: How about: `Enter the dimensions of the 2D array: `?

Also, don't use `endl`, it unnecessary flushes the output.

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 2]
```diff
@@ -43,19 +43,15 @@ void spiralPrint(int a[][100],int m,int n){
...
+    cin >> m >> n;
+    int a[m][n];
```
Comment (arnavb) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: Why not use a nested [`std::vector`](http://en.cppreference.com/w/cpp/container/vector)? Safer and easier to use.

File: code/unclassified/src/spiral_printing/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+-When direction = 3 , we print the left'th column from index bottom to top
+and we increment left by 1
+
+# Example 
+
+Input:
+1  2  3  4
+5  6  7  8
+9  10 11 12
+13 14 15 16
```
Comment (iattempt) [inline code/unclassified/src/spiral_printing/README.md]: Should we change this to:
```
 1  2  3  4  5  6
 7  8  9 10 11 12
13 14 15 16 17 18
```

I think that is more intuitive for output below.

File: code/unclassified/src/spiral_printing/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+# Spiral Printing
+
+The aim is to print a given 2D array in spiral form.
+
+We take the dimensions of the array and the array as an input from the user. Dimensions of matrix are taken as mxn where m are the number of rows and n are the number of columns of the matrix.
+
+To print the matrix in spiral form we first keep a variable direction which can take 4 possible values : 0,1,2,3. 0 means we will print the first row from top which is not printed yet , 1 means first column from right, 2 means first rwo from bottom and 3 means first column from left. We will also be keeping four variables top,bottom,left and right initialised with 0,m-1,0,n-1 respectively which will give us the index of the extremes to be printed.
+
+-When direction = 0 , we print the top'th row from index left to right
```
Comment (arnavb) [inline code/unclassified/src/spiral_printing/README.md]: No  need for the `'th` after each side name.

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 5]
```diff
@@ -43,19 +43,15 @@ void spiralPrint(int a[][100],int m,int n){
...
+    // Take the array as input from the users
+    for(int i=0;i<m;i++)
```
[the remark this answers is not in the document]
Comment (iattempt) [reply code/unclassified/src/spiral_printing/spiral_print.cpp]: also for commas.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**